tests/data_collector/test_data_collector.py

```diff
@@ -141,3 +143,54 @@ def test_truncation_without_reset(dataset_id, env_id):
 
     # check load and delete local dataset
     check_load_and_delete_dataset(dataset_id)
+
+
+@pytest.mark.parametrize("seed", [None, 0, 42, MAX_UINT64])
+def test_reproducibility(seed):
+    """Test episodes are reproducible, even if an explicit reset seed is not set."""
+    dataset_id = "dummy-box-test-v0"
+    env_id = "DummyBoxEnv-v0"
+    policy_seed = 42
+    num_episodes = 5
+
+    env = DataCollector(gym.make(env_id))
+
+    for _ in range(num_episodes):
+        env.reset(seed=seed)
+        env.action_space.seed(policy_seed)
+
+        trunc = False
+        term = False
+
+        while not (trunc or term):
+            _, _, trunc, term, _ = env.step(env.action_space.sample())
+
+    dataset = env.create_dataset(
+        dataset_id=dataset_id,
+        algorithm_name="random_policy",
+        author="Farama",
+        author_email="farama@farama.org",
+    )
+    env.close()
+
+    # Step through the env again using the stored seed and check it matches
+    env = dataset.recover_environment()
+
+    for episode in dataset.sample_episodes(dataset.total_episodes):
+        if seed is None:
+            assert isinstance(episode.seed, int)
+            assert episode.seed >= 0
+        else:
+            assert seed == episode.seed
+
+        obs, _ = env.reset(seed=episode.seed)
+        env.action_space.seed(policy_seed)
+
+        assert np.allclose(obs, episode.observations[0])
+
+        for k in range(episode.total_timesteps):
+            obs, rew, _, _, _ = env.step(env.action_space.sample())
+            assert np.allclose(obs, episode.observations[k + 1])
+            assert rew == episode.rewards[k]
```

Comment:
can you change this loop to:
1) Iterate over the dataset instead of sampling
2) Instead of using `policy_seed` and `env.action_space.sample()`, use the action from the dataset
3) Add also the checks for `termination` and `truncations`